changelog_unreleased/handlebars/pr-7051.md

```diff
@@ -0,0 +1,58 @@
+#### handlebars: Fix superfluous line breaks in ConcatStatement  ([7051](https://github.com/prettier/prettier/pull/7051) by [@dcyriller](https://github.com/dcyriller))
+
+<!-- prettier-ignore -->
+```hbs
+{{!-- input --}}
+<a href="a-very-long-href-from-a-third-party-marketing-platform{{id}}longer-than-eighty-chars">Link</a>
+
+{{!-- Prettier stable --}}
+<a
+  href="a-very-long-href-from-a-third-party-marketing-platform
+    {{id}}
+    longer-than-eighty-chars"
+>
+  Link
+</a>
+
+{{!-- Prettier master --}}
+<a
+  href="a-very-long-href-from-a-third-party-marketing-platform{{id}}longer-than-eighty-chars"
+>
+  Link
+</a>
+```
+
+and
+
+```hbs
+{{!-- input --}}
+<div class="hello {{if goodbye true}} {{if goodbye false}} {{if goodbye true}} {{if goodbye false}} {{if goodbye true}}">
+  Hello
+</div>
+
+{{!-- Prettier stable --}}
+<div
+  class="hello
+    {{if goodbye true}}
+
+    {{if goodbye false}}
+
+    {{if goodbye true}}
+
+    {{if goodbye false}}
+
+    {{if goodbye true}}"
+>
+  Hello
+</div>
+
+{{!-- Prettier master --}}
+<div
+  class="hello {{if goodbye true}} {{if goodbye false}} {{if goodbye true}} {{if
+      goodbye
+      false
+    }} {{if goodbye true}}"
+>
```

Comment:
Maybe better :
```hbs
<div
  class="hello {{if goodbye true}} {{if goodbye false}} {{if goodbye true}} {{
    if goodbye false
  }} {{if goodbye true}}"
>
```

/cc @prettier/core what do you think about that?

Reply:
Also, should:
```
<p>
  Hi {{firstName}} {{lastName}} , welcome!
</p>
```
be formatted to (with a short max-width):
```
<p>
  Hi {{firstName}} {{lastName
  }} , welcome!
</p>
```
or to:
```
<p>
  Hi {{firstName}} {{
    lastName
  }} , welcome!
</p>
```

Reply:
Second example better

Reply:
Here we go: https://github.com/prettier/prettier/pull/7157 🙂

Reply:
Here we go: #7157 🙂